test/lib/check-medic-conf-dependency-version.spec.js

```diff
@@ -0,0 +1,66 @@
+const { expect } = require('chai');
+const sinon = require('sinon');
+const rewire = require('rewire');
+
+const checkMedicConfVersion = rewire('../../src/lib/check-medic-conf-dependency-version');
+const RUNNING_VERSION = '3.1.2';
+
+describe('check-medic-conf-dependency-version', () => { 
+  let warn, fs; 
+  
+  beforeEach(() => {  
+    warn = sinon.stub();
+    fs = {
+      exists: sinon.stub().returns(true),
+      readJson: sinon.stub().returns({ dependencies: { 'medic-conf': '1.0.0' } }),
+    };
+
+    checkMedicConfVersion.__set__('warn', warn);
+    checkMedicConfVersion.__set__('runningVersion', RUNNING_VERSION);
+    checkMedicConfVersion.__set__('fs', fs);
+  });
+
+  const scenarios = [
+    { version: '2.0.0', throw: true },
+    { version: '3.0.0' },    
+    { version: '3.1.1' },
+    { version: '3.1.2' },
+    { version: '3.1.3', throw: true },
```

Comment:
IMO patch version should be fine at any level. If you've previously pushed on 3.1.3 but another person is using 3.1.2 this should be OK, or at worst a warning?

It's important that we block people's progress as infrequently as possible, to avoid error fatigue (see the other comment).

Apologies if this has been discussed and a decision has been made, there is nothing on the ticket so I cannot tell if this has been considered.

Reply:
The reasoning behind not allowing a project version not to be run by an older patch is that there are many releases which fix important breaking bugs which are released as patches and not minor versions e.g https://github.com/medic/medic-conf/commit/eebb5962562d300f6f703f3ed6e156da4e54d7c2 which fixed an important issue in the purging feature.

Reply:
Okey dokey, please put that reasoning on the ticket so we can easily find it later